From 7deb382a96bc6bfbff34b6880106d4875d8baeb8 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Fri, 12 Dec 2025 16:18:15 +0000
Subject: [PATCH] RHEL: output/output.ml: Remove reduce-memory-pressure=on
 (cache=none)

Virt-v2v tries to reduce the amount of page cache used when writing
out the disk image to the target storage. It does this by enabling the
reduce-memory-pressure=on option (previously cache=none) option in
nbdkit-file-plugin.

However, use of nbdkit-file-plugin cache=none option causes data
corruption with Dell PowerMax 8000 storage (only).

As this is only an advisory setting, remove it for now.

Fixes: https://issues.redhat.com/browse/RHEL-135617
---
 output/output.ml | 2 --
 1 file changed, 2 deletions(-)

diff --git a/output/output.ml b/output/output.ml
index ff4b12b1..8b94df34 100644
--- a/output/output.ml
+++ b/output/output.ml
@@ -102,7 +102,6 @@ let output_to_local_file ?name
     | "raw" ->
        let cmd = Nbdkit.create ?name "file" in
        Nbdkit.add_arg cmd "file" filename;
-       Nbdkit.reduce_memory_pressure cmd;
        if verbose () then Nbdkit.add_filter_if_available cmd "count";
        let _, pid = Nbdkit.run_unix socket cmd in
        pid
@@ -209,7 +208,6 @@ let create_local_output_disks dir
     (* Create the single nbdkit-file-plugin instance. *)
     let cmd = Nbdkit.create ~name:"out" "file" in
     Nbdkit.add_arg cmd "dir" output_storage;
-    Nbdkit.reduce_memory_pressure cmd;
     if verbose () then Nbdkit.add_filter_if_available cmd "count";
     let _, pid = Nbdkit.run_unix socket cmd in
     On_exit.kill pid;
